### PR TITLE
#267484 - ImageViewer changes

### PR DESCRIPTION
**Pull request by Chris2011**
* Added new l10n properties for dimension and size.
* Added logic to show dimension and size inside the ImageViewer.

https://netbeans.org/bugzilla/show_bug.cgi?id=267484

### CODE REVIEW

File: image/src/org/netbeans/modules/image/ImageViewer.java [hunk 1]
```diff
@@ -461,10 +486,22 @@ public void run() {
...
+        setLayout(new BorderLayout());
+        add(toolbar, BorderLayout.NORTH);
+    }
+    
     private void showImage(NBImageIcon image, String errorMessage) {
         boolean wasValid = (storedImage != null);
         storedImage = image;
         boolean isValid = (storedImage != null);
+        
+        imageWidth = Integer.toString(storedImage.getIconWidth());
```
Comment (emilianbold) [inline image/src/org/netbeans/modules/image/ImageViewer.java]: You might have a NPE here if isValid == false and storedImage == null, no?

File: image/src/org/netbeans/modules/image/ImageViewer.java [hunk 2]
```diff
@@ -533,9 +571,20 @@ private JToolBar createToolBar() {
 //        sa.putValue (Action.SHORT_DESCRIPTION, NbBundle.getBundle(ImageViewer.class).getString("LBL_CustomZoom"));
         toolBar.add (button = getZoomButton ());
         toolbarButtons.add(button);
+
```
Comment (emilianbold) [inline image/src/org/netbeans/modules/image/ImageViewer.java]: Random whitespace.

File: image/src/org/netbeans/modules/image/ImageViewer.java [hunk 3]
```diff
@@ -127,6 +128,11 @@
     private static final RequestProcessor RP = new RequestProcessor("Image loader", 1, true);
     
     private RequestProcessor.Task loadImageTask;
+    private String imageHeight = "";
+    private String imageWidth = "";
+    private double imageSizeBinary = 0f;
+    private double imageSizeDecimal = 0f;
+    private String imageSizeLabel = "LBL_ImageSizeBytes";
```
Comment (emilianbold) [inline image/src/org/netbeans/modules/image/ImageViewer.java]: Add //NOI18N

File: image/src/org/netbeans/modules/image/ImageViewer.java [hunk 4]
```diff
@@ -152,21 +158,40 @@ protected void reloadIcon() {
...
+        if (imageFile.isValid()) {
+            imageSizeBinary = imageFile.getSize();
+            double binaryPrefix = 1024f;
+            double decimalPrefix = 1000f;
+
+            // KB to MB
+            if(imageSizeBinary >= (binaryPrefix * binaryPrefix)) {
+                imageSizeDecimal = roundToTwoDecimalPlaces(imageSizeBinary / (decimalPrefix * decimalPrefix));
+                imageSizeBinary = roundToTwoDecimalPlaces(imageSizeBinary / (binaryPrefix * binaryPrefix));
+                imageSizeLabel = "LBL_ImageSizeMb";
```
Comment (emilianbold) [inline image/src/org/netbeans/modules/image/ImageViewer.java]: Add //NOI18N

File: image/src/org/netbeans/modules/image/ImageViewer.java [hunk 5]
```diff
@@ -152,21 +158,40 @@ protected void reloadIcon() {
...
+              // Bytes to KB
+            } else if(imageSizeBinary >= binaryPrefix) {
+                imageSizeDecimal = roundToTwoDecimalPlaces(imageSizeBinary / decimalPrefix);
+                imageSizeBinary = roundToTwoDecimalPlaces(imageSizeBinary / binaryPrefix);
+                imageSizeLabel = "LBL_ImageSizeKb";
```
Comment (emilianbold) [inline image/src/org/netbeans/modules/image/ImageViewer.java]: Add //NOI18N

File: image/src/org/netbeans/modules/image/ImageViewer.java (hunk 3 above)
Comment (Chris2011) [inline image/src/org/netbeans/modules/image/ImageViewer.java]: Only to the last line or to 132 - 135? I think only to the last right?

File: image/src/org/netbeans/modules/image/ImageViewer.java (hunk 2 above)
Comment (Chris2011) [inline image/src/org/netbeans/modules/image/ImageViewer.java]: Yeah well, it was not random. It is intended. I wanted to make it more readable the whole code, I know it isn't I separated the code what has nothing todo with the lines above and that I changed. I sure I can redo it but it shouldn't matter. I will be more careful in the future to not reformat code.

File: image/src/org/netbeans/modules/image/ImageViewer.java (hunk 1 above)
Comment (Chris2011) [inline image/src/org/netbeans/modules/image/ImageViewer.java]: I moved the code to the isValid check, will commit it soon.

**Comment by Chris2011** [comment]
Hopefully everything is fine now ;) :D

**Comment by emilianbold** [comment]
Pushed as https://hg.netbeans.org/jet-main/rev/1b877de38127

**Pull request closed without merging**